src/base/bittorrent/categoryoptions.cpp

```diff
@@ -31,13 +31,24 @@
 #include <QJsonObject>
 #include <QJsonValue>
 
-const QString OPTION_SAVEPATH {QStringLiteral("save_path")};
+const QString OPTION_SAVEPATH {QStringLiteral("savePath")};
 const QString OPTION_DOWNLOADPATH {QStringLiteral("download_path")};
 
 BitTorrent::CategoryOptions BitTorrent::CategoryOptions::fromJSON(const QJsonObject &jsonObj)
 {
     CategoryOptions options;
-    options.savePath = jsonObj.value(OPTION_SAVEPATH).toString();
+
+    const auto savePathIter = jsonObj.constFind(OPTION_SAVEPATH);
+    if (savePathIter != jsonObj.constEnd())
+    {
+        options.savePath = savePathIter->toString();
+    }
+    else
+    {
+        // TODO: remove this fallback code when qbt >= v4.6.0
+        // this fallback is for wrong key briefly used in in v4.4.0
```

Comment:
This key isn't "wrong". This serialization was designed for core purposes. It was a mistake to use it in WebAPI as well.
Every time I face a dilemma, whether to have a separate serialization for WebAPI (which is more flexible, but often duplicates something else), or to use the same one with the core (which complicates code maintenance).

Reply:
Anyway keys should be consistent.

Reply:
Changed.
It is strange that you used underline for keys...

Reply:
> It is strange that you used underline for keys...

For some reason, it doesn't look so bad when used in JSON, file names etc. But I don't object to change them here.

P.S. the inconsistent keys are still in WebAPI.

Reply:
>P.S. the inconsistent keys are still in WebAPI.

I don't understand.

Reply:
> > P.S. the inconsistent keys are still in WebAPI.
> 
> I don't understand.

I mean key names:
`savePath` and `download_path`.

Reply:
>For some reason, it doesn't look so bad when used in JSON, file names etc. But I don't object to change them here.

I don't want to change it unless it is really necessary.

>I mean key names:

I see, I hope adjusting them at the edge won't become a maintenance burden in the long run.

Reply:
>I mean key names:

Well, the situation is more twisted than I thought. There are usage of `download_path` in other WebAPI code. And now in category action we used `downloadPath`. Isn't that inconsistent?
I can think of 2 ways to resolve:
1. Change all webUI code to use the new key `save_path`. I don't really like this webAPI breakage and changing it is error prone.
2. Use `savePath` and leave `download_path` as is. But I guess you won't like it.

Reply:
One more question: How about those who changed their apps to follow current API changes? Maybe keep it as is and just add savePath for backward compatibility. 

All those who want to have access to new features (like per-torrent download path) in their client applications will have to make appropriate changes there, so it's not a problem for them to change `savePath` to `save_path` as well (`save_path` should be recommended).

Reply:
>One more question: How about those who changed their apps to follow current API changes? Maybe keep it as is and just add savePath for backward compatibility.

After seeing the user complaints for 4.4.0, I don't think the majority will be using it. IMO it would be ok if we just admit faults and say it is an unwanted change.

>(save_path should be recommended).

What about WebUI? I don't really want to delve in... if you want to change it then I'll leave the whole thing to you (closing this PR).

Currently I still favor option 2.

Reply:
> After seeing the user complaints for 4.4.0, I don't think the majority will be using it. IMO it would be ok if we just admit faults and say it is an unwanted change.

Ok.

Reply:
>Ok.

Hope that means you are ok with option 2.
Now I pushed the intended fix.

Reply:
> > Ok.
> 
> Hope that means you are ok with option 2. Now I pushed the intended fix.

As you want. Talking about the consistency of what is mostly inconsistent is not a very good idea. WebAPI cannot be helped without a total sweep, but this is not a priority task now.